lib/topology/platforms/shell.py

```diff
@@ -328,31 +338,39 @@ def _get_connect_command(self):
         :return: The command to be used when connecting to the shell.
         """
 
+    def _get_connection(self, connection):
+        """
+        Get the pexpect object for the given connection name.
+
+        :param str connection: Name of the connection.
+        :rtype: :class:`pexpect.spawn`
+        :return: The PExpect spawn process that handles the connection.
+        """
+        connection = connection or self._default_connection or '0'
+
+        if connection not in self._connections:
+            raise NonExistingConnectionError(connection)
+
+        return self._connections[connection]
+
     def send_command(
-        self, command, matches=None, newline=True, timeout=None,
-        connection=None
+        self, command,
+        matches=None, newline=True,
+        timeout=None, connection=None
     ):
         """
         See :meth:`BaseShell.send_command` for more information.
         """
-        # Connection is undefined
-        if connection is None:
-            # Default connection is not connected
-            if not self.is_connected(self._default_connection):
-                # Connect the default connection
-                self.connect(connection)
-            # Either way, an unspecified connection means default connection
-            connection = self._default_connection
-
-        # Connection is defined but nonexistent
-        elif connection not in self._connections:
-            self.connect(connection)
+        # Get connection
+        try:
+            spawn = self._get_connection(connection)
 
-        # Connection is defined and existent
-        else:
-            # Connect if disconnected
-            if not self.is_connected(connection):
-                self.connect(connection)
+        except NonExistingConnectionError:
```

Comment:
The previous behavior of connecting a previously disconnected shell is not kept because of this. An existing but disconnected connection won't be connected when calling this method.